api/apps/geoprocessing/src/modules/planning-area/planning-area.service.ts

```diff
@@ -1,23 +1,46 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 import { GeoJSON } from 'geojson';
 import { CustomPlanningAreaRepository } from '@marxan/planning-area-repository';
 import { ShapefileService } from '@marxan-geoprocessing/modules/shapefiles/shapefiles.service';
+import { GarbageCollectorConfig } from '@marxan-geoprocessing/modules/planning-area/garbage-collector-config';
 
 @Injectable()
 export class PlanningAreaService {
+  private readonly logger = new Logger(this.constructor.name);
+
   constructor(
     private readonly repository: CustomPlanningAreaRepository,
     private readonly shapefileService: ShapefileService,
+    private readonly config: GarbageCollectorConfig,
   ) {}
 
   async save(
     shapefile: Express.Multer.File,
   ): Promise<{ data: GeoJSON; id: string }> {
     const { data } = await this.shapefileService.transformToGeoJson(shapefile);
     const result = await this.repository.saveGeoJson(data);
+    this.scheduleGarbageCollection();
     return {
       id: result[0].id,
       data,
     };
   }
+
+  private scheduleGarbageCollection() {
+    setTimeout(async () => {
+      this.logger.debug(`garbage collection scheduled`);
+      const removedCount = await this.collectGarbage();
+      this.logger.debug(
```

Comment:
very minor, but if we want to keep some logging output, I suggest to make this `.log()`, to pass in the max age value and to output this too (`...removed <count> entities older than <age>`).

We already have some time diff in the default logs to know how long the GC takes.

I would be tempted to suggest to add a `v4()` to the two log lines in case we have >1 GC ops triggered at the kind of exact same time (two users saving planning areas at the same time) but I'd say it's very unlikey and the utility we could get from knowing for sure which of the two (or more 😁) GC ops started and ended when it's very low.
Feel free to leave this as is, in any case - since we already have some output, I just suggested to make it more informative with age data, in case we need to check back anything in logs